datafusion/functions-aggregate/src/nth_value.rs

```diff
@@ -430,3 +428,176 @@ impl NthValueAccumulator {
         Ok(())
     }
 }
+
+/// This is a wrapper struct to be able to correctly merge `ARRAY_AGG` data from
```

Comment:
Beyond this point is entirely [duplicated code from `array_agg_ordered`](https://github.com/apache/datafusion/blob/7df000a333a7d4018e1446ef900f652288b1f104/datafusion/physical-expr/src/aggregate/array_agg_ordered.rs#L387)

Can be removed once `array_agg_ordered` is also converted.

Reply:
Alternative way is move this to physical-common, after all the function are cleanup, we can move it back to functions-aggregate crate

Reply:
Done in https://github.com/apache/datafusion/pull/11287/commits/6ce667939b8a0bc05b143d6c653f167fcc4ea639.

The duplication of code is now eliminated. Agreed, at the end we can move this back into functions-aggregate.